extract/smil-data.es6

```diff
@@ -11,29 +12,28 @@ const VERSION = 'version';
 export default function smilData(xml, manifestItem, spineItem, {clockValue}) {
   const smilXml = xml.querySelector(TAG);
   const children = extractChildren(smilXml);
-  const smilVersion = smilXml.getAttribute(VERSION);
+  const version = smilXml.getAttribute(VERSION);
 
   let ret = {
     children,
-    href: manifestItem.href,
     id: manifestItem.id,
-    spineItemId: spineItem.id,
-    smilVersion
+    version
   };
 
   if (clockValue) {
     ret.duration = parseClockValue(clockValue);
   }
-
   return ret;
 }
 
 function extractChildren(xml) {
-  return Array.prototype.map.call(xml.childNodes, extractAttributes).filter(c => !!c);
+  const filteredNodes = Array.prototype.filter.call(xml.childNodes, canIgnoreNode);
```

Comment:
Once you do a filter you already have an array, so you can just use map over it. Then the code above becomes:

```
return Array.prototype.filter.call(xml.childNodes, canIgnoreNode).map(extractAttributes);
```